classes/tools/google/seo.php

```diff
@@ -48,80 +47,150 @@ public static function getAnalyticsTrackingScript() {
                 $view = 'bru_google_seo_tools::js_tag_universal_analitycs';
                 $datas = array(
                     'tag' => $tag,
-                    'domain' => self::getDomain($current_context),
+                    'domain' => self::getDomain(),
                 );
             } else {
                 $view = 'bru_google_seo_tools::js_tag';
                 $datas = array(
                     'tag' => $tag,
                 );
             }
-            $full_script = \View::forge($view, $datas);
+            $fullScript = \View::forge($view, $datas);
         }
 
-        if ($full_script === '') {
+        $fullScript = static::getCodeWithoutComment($fullScript);
+
+        if (empty($fullScript)) {
+            return '';
+        } else {
+            self::_doNotTrack($fullScript);
+
+            return (string) $fullScript;
+        }
+    }
+
+    /**
+     * Returns the body script tagmanager wich contain the google analytics javascript
+     *
+     * @return string
+     */
+    public static function getTagmanagerTrackingScriptBody()
+    {
+        $config = static::getConfig();
+
+        if (empty($config)) {
+            return '';
+        }
+
+        if ($config['tagmanager_full_script_body'] != '') {
+            if (\Str::starts_with($config['tagmanager_full_script_body'], '<script')) {
+                $fullScript = $config['tagmanager_full_script_body'];
+            } else {
+                $fullScript = '<script type="text/javascript">'.$config['tagmanager_full_script_body'].'</script>';
+            }
+        } else {
+            $tag = \Arr::get($config, 'google_tagmanager_tag');
+            $fullScript = \View::forge('bru_google_seo_tools::js_tagmanager_body', compact('tag'));
+        }
+
+        $fullScript = static::getCodeWithoutComment($fullScript);
+
+        if (!empty($fullScript)) {
+            self::_doNotTrack($fullScript);
+        }
+
+        return (string) $fullScript;
+    }
+
+    /**
+     * Returns the head script tagmanager wich contain the google analytics javascript
+     *
+     * @return string
+     */
+    public static function getTagmanagerTrackingScriptHead()
+    {
+        $config = static::getConfig();
+
+        if (empty($config)) {
             return '';
+        }
+
+        if ($config['tagmanager_full_script_head'] != '') {
+            if (\Str::starts_with($config['tagmanager_full_script_head'], '<script')) {
+                $fullScript = $config['tagmanager_full_script_head'];
+            } else {
+                $fullScript = '<script type="text/javascript">'.$config['tagmanager_full_script_head'].'</script>';
+            }
         } else {
-            self::_doNotTrack($full_script);
-            return (string)$full_script;
+            $tag = \Arr::get($config, 'google_tagmanager_tag');
+            $fullScript = \View::forge('bru_google_seo_tools::js_tagmanager_head', compact('tag'));
```

Comment:
Same as above, third parameter to false on View::forge() to prevent auto-escape ?